lib/tokenizer.js

```diff
@@ -40,26 +42,48 @@ class Tokenizer {
   /**
    * Parses a token string and returns a representation of the token
    * @abstract
-   * @param {String} value
+   * @param {Token} value
    */
   parse(value) {
     throw new Error('You must implement a parse function for the tokenizer');
   }
 
+  minToken() {
+    throw new Error('You must implement a minToken function for the tokenizer');
+  }
+
   /**
-   * Returns 0 if the values are equal, 1 if val1 is greater then val2 and -1 if val2 is greater than val1
-   * @param val1
-   * @param val2
-   * @returns {number}
+   * Splits the range specified by start and end into numberOfSplits equal parts.
+   * @param {Token} start Starting token
+   * @param {Token} end  End token
+   * @param {Number} numberOfSplits Number of splits to make.
    */
-  compare(val1, val2) {
-    if (val1 > val2) {
-      return 1;
-    }
-    if (val1 < val2) {
-      return -1;
+  split(start, end, numberOfSplits) {
+    throw new Error('You must implement a split function for the tokenizer');
+  }
+
+  splitBase(start, range, ringEnd, ringLength, numberOfSplits) {
```

Comment:
Can we add jsdoc comments to include the param types (`Integer`) for future reference?